### ISSUE TITLE
Linux/Reaper crashed when adding the plugin a 2nd time

### ISSUE THREAD

**Issue by maxxatgit** [ardura/Subhoofer#9]
I'm running Ubuntu 22.04 and Reaper 7.16. 
To reproduce the crash:

1. I open Reaper
2. add either the .clap or the .vst3 version to a project
3. remove the plugin from the project
4. Attempt to add it again

At this point, Reaper freezes and the popup "reaper not responding" comes up. I have to select "force close".
Just as an FYI, the NIH Spectral Compressor by Robert-VDH (another rust-based plugin) runs fine on my system.

If there is anything that I can do to help find this bug, please let me know.
Thanks for supporting Linux!

**Comment by ardura** [comment]
Hey maxxatgit! Hmm that does sound weird...I'll see if I can spin up a linux machine as well and try it out but otherwise I may have to finally make a debugging build for issues like this. I'll see what I can find then update you!

**Comment by ardura** [comment]
I'm unable to replicate this so far, I am able to load,unload,and reload CLAP and VST versions of Subhoofer on my linux (Crostini Debian) and Reaper 7.16. However I will note that the precompiled builds didn't work on my test system and I compiled it myself for them to work (otherwise they wouldn't even load). Are you compiling Subhoofer or using the precompiled binary?

**Comment by ardura** [comment]
Here are updated instructions in case you are compiling, although the github runner builds the precompiled binary on Ubuntu 22.04 as well... so that would also be very strange. Is there anything in your Reaper logs or something like that? I couldn't really find much on debugging vsts in Reaper so far.

https://github.com/ardura/Subhoofer?tab=readme-ov-file#buildingcompiling-subhoofer-manually

**Comment by maxxatgit** [comment]
I am using the precompiled binary. Unfortunately, there is nothing in the Reaper logs.
Sometimes opening Reaper from a terminal window will give pertinent info as it exits. With this issue, it just says "killed" (probably because I click the "force quit" button).
I don't want to start compiling plugins on my production machine. Maybe KXStudio or OpenSUSE will do it.

**Comment by maxxatgit** [comment]
I would be happy to run a debug version and report back.

**Comment by ardura** [comment]
Fair enough 👍 I'll work on making a debug build happen. I'll probably update this next week and then we can try to figure this out

**Comment by maxxatgit** [comment]
Greetings, and thanks for creating a debug build. The crash happened on the
third attempt to load the clap version.
add: OK
remove: OK
add: OK
remove: OK
add: crash!

Here is the contents of subhoofer.log:

(emacs:15336): Gtk-WARNING **: 19:24:40.781: Could not load a pixbuf from
/org/gtk/libgtk/icons/16x16/status/image-missing.png.

This may indicate that pixbuf loaders or the mime database could not be
found.

**

Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon:
assertion failed (error == NULL): Failed to load
/org/gtk/libgtk/icons/16x16/status/image-missing.png: Unrecognized image
file format (gdk-pixbuf-error-quark, 3)

Bail out!
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon:
assertion failed (error == NULL): Failed to load
/org/gtk/libgtk/icons/16x16/status/image-missing.png: Unrecognized image
file format (gdk-pixbuf-error-quark, 3)

Fatal error 6: Aborted

Backtrace:

/snap/emacs/2464/usr/bin/emacs-gtk(+0x1b0598)[0x5cebb77de598]

/snap/emacs/2464/usr/bin/emacs-gtk(+0x55871)[0x5cebb7683871]

/snap/emacs/2464/usr/bin/emacs-gtk(+0x55dbc)[0x5cebb7683dbc]

/snap/emacs/2464/usr/bin/emacs-gtk(+0x1ae5bd)[0x5cebb77dc5bd]

/snap/emacs/2464/usr/bin/emacs-gtk(+0x1ae6af)[0x5cebb77dc6af]

/snap/emacs/2464/usr/bin/../../lib/x86_64-linux-gnu/libpthread.so.0(+0x14420)[0x77a2589f0420]

/snap/emacs/2464/usr/bin/../../lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x77a256a1800b]

/snap/emacs/2464/usr/bin/../../lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x77a2569f7859]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x1db43)[0x77a25a11fb43]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libglib-2.0.so.0(g_assertion_message_error+0xb6)[0x77a25a17d4c6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x213bca)[0x77a25aa60bca]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x214278)[0x77a25aa61278]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x214394)[0x77a25aa61394]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2144e0)[0x77a25aa614e0]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x228191)[0x77a25aa75191]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x182417)[0x77a25a9cf417]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18675e)[0x77a25a9d375e]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x228927)[0x77a25aa75927]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ac3)[0x77a25ab36ac3]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ce6)[0x77a25ab36ce6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0xa5)[0x77a25ab37035]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x12fd23)[0x77a25a97cd23]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x182417)[0x77a25a9cf417]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18675e)[0x77a25a9d375e]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x1306a7)[0x77a25a97d6a7]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ac3)[0x77a25ab36ac3]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ce6)[0x77a25ab36ce6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0xa5)[0x77a25ab37035]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x182417)[0x77a25a9cf417]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x18675e)[0x77a25a9d375e]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x13d607)[0x77a25a98a607]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ac3)[0x77a25ab36ac3]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ce6)[0x77a25ab36ce6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0xa5)[0x77a25ab37035]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ac3)[0x77a25ab36ac3]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ce6)[0x77a25ab36ce6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_preferred_height+0x89)[0x77a25ab36179]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e94fb)[0x77a25ab364fb]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2e9ce6)[0x77a25ab36ce6]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(gtk_widget_get_preferred_height_and_baseline_for_width+0xa5)[0x77a25ab37035]

/snap/emacs/2464/usr/bin/../lib/x86_64-linux-gnu/libgtk-3.so.0(+0x2ea18b)[0x77a25ab3718b]

...

Aborted

**Comment by maxxatgit** [comment]
Greetings.
I have to do something different. That is just an emacs error. It has
nothing to do with the plugin.
More info soon...

**Comment by maxxatgit** [comment]
OK, the log file is empty. zero bytes long.
Reaper crashes with no message in the terminal.

**Comment by maxxatgit** [comment]
Those are the normal (for me) settings for VST plugins...
I tried the VST3 version, and got the same results.

I also tried to make the NIH Spectral Compressor crash, and I had no problem with it.
I opened and closed it 4 or 5 times without issues.

One thing that may be a factor, I'm using an AMD Ryzen 9 7950X with an integrated GPU. I'm using the internal GPU, not a card. I've heard that the make and model of GPU can cause issues... 
Also, I'm running the Liquorix kernal...

I can switch to the standard kernal to test sometime next week. I don't have time to get into all of that right now.

I appreciate your diligence, but don't stress about this.

As always, thank you for supporting Linux :)

**Comment by maxxatgit** [comment]
So it seems @PolyVector found that the bug resides in Rust Audio. And their comment in the linked thread, "Because it's a race condition that also depends on driver quirks, every system I've tested behaves differently. Usually crashes occur upon reopening the window. Some systems crash on the first reopening, some can survive 100, it's all over the place." is the reason I didn't notice it in the NIH plugins...
:^)

**Comment by ardura** [comment]
@maxxatgit dang, that makes sense why I couldn't figure it out either. At least they're aware of it in the Nih-Plug side. Sorry there's no current fix though 🤔 if one does appear I'll update the libs to fix it 👍

**Comment by ardura** [comment]
Since that bug has been there since 2023...I took a look back at that linked issue and it seems like something I can try to implement in my baseview/egui-baseview (whichever one) fork that my plugins use. Will update you once I do that and we can test again :)

**Comment by maxxatgit** [comment]
Yeah, checking with the baseview maintainers sounds like a good idea. @PolyVector did mention that their fix was temporary, that "A proper fix would be to synchronize with the event loop thread". It seems like that may take a while, so it would be good to know how the maintainers plan to respond to this.

Thanks for all you do!

**Comment by maxxatgit** [comment]
Good news!
The issue in baseview has been closed!
I wanted to check here before I tested the latest build, since it didn't reference this issue in the change log.

**Comment by ardura** [comment]
@maxxatgit Can you try with this build and let me know if the issue is fixed?

[Subhoofer_v223_windows_build.zip](https://github.com/user-attachments/files/24436973/Subhoofer_v223_windows_build.zip)
[Subhoofer_v223_macos_m1_build.zip](https://github.com/user-attachments/files/24436972/Subhoofer_v223_macos_m1_build.zip)
[Subhoofer_v223_macos_build.zip](https://github.com/user-attachments/files/24436971/Subhoofer_v223_macos_build.zip)
[Subhoofer_v223_linux_build.zip](https://github.com/user-attachments/files/24436970/Subhoofer_v223_linux_build.zip)

**Comment by ardura** [comment]
That's great news! I'll push this test branch into the main one then. Thanks for the help figuring this out.

**Comment by RustoMCSpit** [comment]
added to OAS https://github.com/open-audio-stack/open-audio-stack-registry/issues/187
